Develop

HaliGali101 opened the pull request.

HaliGali101 committed:
develop/
проект 7 спринта предварительная версия

HaliGali101 committed:
develop/
allure отчёт

HaliGali101 committed:
develop/
финальная версия проекта

HaliGali101 committed:
allure отчёт

egorvas commented on src/main/java/scooter/testData/OrdersData.java:
```diff
@@ -0,0 +1,22 @@
+    public static Orders getOrderData(ArrayList<String> colorsArray) {
+        String firstName = "Тестовый";
+        String lastName = "Тест";
+        String address = "Москва, пр.Мира д.120";
```
**Можно лучше:** Вместо того чтобы самому придумывать тестовые строки я рекомендую использовать модули для генерации данных, например java faker: https://github.com/DiUS/java-faker
С его помощью ты сможешь убрать большую часть хардкода из твоих тестов.

HaliGali101 replied:
исправлено в e2addee0108c8826a02bd2e89b73fce90be6ef28

egorvas commented on src/test/java/scooter/OrdersListTests.java:
```diff
@@ -0,0 +1,99 @@
+    public void ordersListTest() {
+        getOrdersList()
+                .statusCode(200)
+                .assertThat().body("orders", notNullValue());
```
**Можно лучше:** notNullValue это крайне слабая проверка. Если ответ будет {"orders":"error"} то твой тест пройдет

egorvas commented on src/test/java/scooter/OrdersListTests.java:
```diff
@@ -0,0 +1,99 @@
+    }
+
+    @Step("Получить id заказа")
+    public void getOrderInfo() {
```
**Можно лучше:** степы лучше выносить в отдельные классы, чтоб не было такого что один тест ссылается на степы другого

egorvas commented on src/test/java/scooter/CreateCourierTests.java:
```diff
@@ -0,0 +1,128 @@
+    }
+
+    @Step("Удалить курьера")
+    private ValidatableResponse deleteCourier(String id) {
```
**Можно лучше:** такой степ точно стоит вынести в отдельный класс, чтоб можно было переиспользовать

egorvas commented on src/test/java/scooter/OrderTests.java:
```diff
@@ -0,0 +1,81 @@
+    @Test
+    @DisplayName("Создание заказа.Чёрный цвет")
+    @Description("Создать заказ с чёрным цветом самоката")
+    public void createOrderWithBlackColor() {
```
**Нужно исправить:** по условиям задания "Чтобы протестировать создание заказа, нужно использовать параметризацию." У тебя я параметризацию не вижу

HaliGali101 replied:
Исправлено в f08ba99301745c37bb0ca7d1592f83065318d02b

HaliGali101 committed:
develop/
добавлена параметризация в тестовый класс OrderTests.java, согласно условию задачи

HaliGali101 committed:
develop/
исправлена генерация объекта заказа при помощи javafaker

HaliGali101 committed:
develop/
добавлена валидация ответа по схеме в OrdersListTests.java с помощью библиотеки json-schema-validator

HaliGali101 committed:
develop/
все шаги тестов вынесены в отдельные классы

HaliGali101 committed:
develop/
добавлен новый результат прогона allure
